### PR TITLE
pin-2013: Auto approve if the consumer is the producer

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 1]
```diff
@@ -460,9 +458,16 @@ final case class AgreementApiServiceImpl(
       case AgreementState.ACTIVE  =>
         Future.successful(agreement.stamps.copy(submission = stamp.some, activation = stamp.some))
       case AgreementState.MISSING_CERTIFIED_ATTRIBUTES => Future.successful(agreement.stamps)
-      case _ => Future.failed(AgreementNotInExpectedState(agreement.id.toString(), newState))
+      case _ => Future.failed(AgreementNotInExpectedState(agreement.id.toString, newState))
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: deprecated! :D

Comment (galales) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: not in our version :D

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala (hunk 1 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: 😡 x 100

Comment (galales) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala]: I'll put it back just for you

**Pull request merged**